content/posts/fission_builder_podspec.md

```diff
@@ -0,0 +1,181 @@
+---
+title: "Function builders also support PodSpec now"
+date: 2019-11-01T12:39:32+05:30
+draft: false 
+author: "[Vivek Singh](https://twitter.com/viveksinghggits)"
+---
+
+In a previous [post](https://blog.fission.io/posts/fission_pod_specs/) we discussed about how we can leverage PodSpec in the environment pods to enable the functionalities like tolerations, nodeSelectors, volumes, security and a lot others. That functionality was only supported in the environment pods, but now we can provide the `podSpec` in the builder pods as well. The details about what podspec is and the functionalities that it enables are mentioned in the mentioned post. 
+
+In this post we will be looking into how do we use podspec in builder so that the deployment that will be created for the builder will have podspec that we mention in the environment for builder. If we take a look at below environment spec
+{{< highlight "yaml" >}}
+apiVersion: fission.io/v1
+kind: Environment
+metadata:
+  creationTimestamp: null
+  name: python-env
+  namespace: default
+spec:
+  builder:
+    command: build
+    image: fission/python-builder
+  keeparchive: false
+  poolsize: 3
+  resources: {}
+  runtime:
+    image: fission/python-env
+  terminationGracePeriod: 360
+  version: 2
+{{< /highlight  >}}
+
+ 
+We can see that for builder we are just specifying the command and image, and if we look into the pod that will be created, if we apply this spec, it will only have standard podspec that will be set by fission itself.
+
+## Using NodeSelector in builder podspec
+Now assume that you want the builder pod for this fission environment to only get scheduled on the machine that has some label, let's say `machinecap=high`, till now there was no way to do that but now you can have podspec in the builder itself to achieve the same. Now lets have a look at the environment spec below
+
+{{< highlight "yaml" >}}
+apiVersion: fission.io/v1
+kind: Environment
+metadata:
+  creationTimestamp: null
+  name: py-builder
+  namespace: default
+spec:
+  builder:
+    command: build
+    image: fission/python-builder
+    container: 
+      command: ["sleep", "1"]
+    podspec:
+      containers:
+      - name: containertwo 
+        image: nginx 
+      nodeSelector: 
+        machinecap: high 
+  keeparchive: false
+  poolsize: 3
+  resources: {}
+  runtime:
+    image: fission/python-env
+    terminationGracePeriod: 360
+  version: 2
+{{< /highlight  >}}
+
+You can see that for builder we are now specifying the podspec and in that we can have almost everything that is supported in kuberentes podspec specification, for example to satisfy our requirement to schedule the builder pod only on the nodes that have label `machinecap=high` we have introduced [nodeSelector](https://kubernetes.io/docs/concepts/configuration/assign-pod-node/#nodeselector) in the builder podspec that takes care of this requirement.
```

Comment:
s/kuberentes podspec specification/Kubernetes PodSpec specification/ 
Might want to add link to API doc here? https://v1-16.docs.kubernetes.io/docs/reference/generated/kubernetes-api/v1.16/#podspec-v1-core